proposals/records-wip.md

```diff
@@ -99,3 +99,4 @@ of the left-hand side.
 The evaluation of a `with` expression is equivalent to calling the `Clone` method exactly once,
 and then setting the backing field of each record property in the argument list to its corresponding
```

Comment:
>backing field [](start = 21, length = 13)

The backing field is not typically accessible to the caller.

Reply:
It does if it's generated!

Reply:
Not according to the spec above.  Per spec, an auto-property is generated, and there is no accessible backing field in that case.

---
In reply to: [406952450](https://github.com/dotnet/csharplang/pull/3353#discussion_r406952450) [](ancestors = 406952450)

Reply:
There is a backing field, and the semantics are defined st. it were as though the backing field were set. There's no requirement that the operation being performed is a legal C# assignment, so there's no requirement that the backing field is accessible.

Reply:
You probably want to say some of this in the spec.

---
In reply to: [406956451](https://github.com/dotnet/csharplang/pull/3353#discussion_r406956451) [](ancestors = 406956451)

Reply:
> You probably want to say some of this in the spec. 

Fair! I'll try to clarify that where some of the existing language rules are being bypassed.

> Moreover, it isn't clear how this could be mapped onto verifiable IL.

IL verifiability isn't well-defined in the C# language specification, so I'll leave that for future work.

Reply:
When the spec defines something as "equivalent to" something else, that includes legality of doing that something else in C#.  If instead you're defining the runtime semantics, just say that the backing field is assigned (no "equivalence" is involved).

---
In reply to: [406956451](https://github.com/dotnet/csharplang/pull/3353#discussion_r406956451) [](ancestors = 406956451)

Reply:
As I understand the initonly feature, its backing field cannot be set directly by the caller. But its setter can be called during the initialization phase. If you intend the `with` expression to be part of the initialization phase, you need to specify where it begins, what value's initialization field it is the initialization phase *for*, and where it ends.

---
In reply to: [406950140](https://github.com/dotnet/csharplang/pull/3353#discussion_r406950140) [](ancestors = 406950140)